vimiv/api/__init__.py

```diff
@@ -94,5 +94,6 @@ def add_external_format(
         test_func: Function returning True if load_func supports this type.
         load_func: Function to load a QPixmap from the passed path.
     """
-    files.add_image_format(file_format, test_func)
+    # Priority to overwrite potential "default behaviour"
+    imageheader.register(file_format, test_func, priority=True)
```

Comment:
The priority is needed as some CR2 used to be detected as JPG or something funky I assume? Just curious here :blush:

Reply:
That is actually a very good point. I must have made an error in thought; The order of the check function registry has obviously no impact on the priority of QPixmap handler function in `utils.imagereader` 🙈

The second reason, which is actually still valid, is that this improves the performance of explicitly registered formats, as they are checked before all the others.

Collisions should theoretically not really happen. Compared to `imghdr` it should be even less likely, as they did sometimes some shortcuts for the checks that I did not do.

So, should I maintain this option, or change something?

Reply:
It does define the QPixmap handler function though, as we pass the `file_format`, or am I having a brainfreeze?
```python
        self._handler = QImageReader(path, file_format.encode())
```

I am totally fine with keeping this, just thought we actually had some explicit reason back then as well, maybe I am wrong.

Reply:
Actually you are right! The issue was that CR2s were detected as TIFFs (because it is based on TIFF), and then the wrong handler was used. You have such a great memory 😳🫣

While my TIFF detection is certainly better than imghdr's (uses twice the amount of information; as said, they sometimes do some simplifications), it unfortunately still matches the beginning of CR2's signature.

I have improved to comment. In addition I have prioritized the external reader in `utils.imagereader` over the base reader. This ensures that when a user registered a external handler for a format what is already supported by QT (e.g. to change some behaviour in how the image is loaded/rendered), the explicitly registered handler is used.
This is in some way similar to how the external format is prioritized over a qt format in case of a signature collision. Let me know if this makes sense to you.

See 60317ddf7752d6deb2281526f974b51b11936b89

Reply:
Makes perfect sense, thanks for checking and updating! :blush: